File: api/Models/Season.cs
using System.Text.Json.Serialization;

namespace api.Models
{
    public class Season
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = default!;
        public string? Description { get; set; }
        public bool IsArchived { get; set; }
        public DateTime CreatedAt { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public List<Permission> Permissions { get; set; } = default!;
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public List<Team>? Teams { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public List<Driver>? Drivers { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public List<Race>? Races { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public List<Favorite> Favorites { get; set; } = default!;
    }
}

File: web/src/app/models/season.ts
import { Driver } from "./driver";
import { Race } from "./race";
import { Team } from "./team";
import { Permission } from "./permission";

export class Season {
  constructor(
    public id: string,
    public name: string,
    public description: string,
    public isArchived: string,
    public createdAt: Date,
    public favorite: number,
    public permissions: Permission[],
    public teams: Team[],
    public drivers: Driver[],
    public races: Race[]
  ) {}
}
